cmd/manager/main.go

```diff
@@ -215,6 +217,7 @@ func addMetrics(ctx context.Context, cfg *rest.Config) error {
 	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
 	if err != nil {
 		log.Info("Could not create metrics Service", "error", err.Error())
+		return err
```

Comment:
We'd only want this to return an error if the metrics service couldn't be created and the reason **wasn't** because it already existed.

So this would require some examination of the error returned by CreateMetricsService() to see what the error looks like when the service already exists. Having had a quick glance at the CreateMetricsService code, this might be a little awkward, because it doesn't pass the k8s error up the chain, it wraps it in another error.

What I would recommend is running the operator on a cluster in debug mode (check out Eric's video - I'll send a link) and seeing what error is returned when the service already exists. Then writing a code block accordingly such that "if the error isn't like this, then return an error".